native/src/routes/landing/containers/LandingContainer.js

```diff
@@ -62,22 +62,18 @@ class LandingContainer extends React.Component<ContainerPropsType> {
     const path = `/${cityCode}/${language}`
     const key: string = generateKey()
 
+    navigation.navigate('CityContent')
+
     // $FlowFixMe newKey missing in typedef
-    const action: NavigationReplaceAction = StackActions.replace({
+    navigation.dispatch(StackActions.replace({
       routeName: 'Dashboard',
       params: {
         cityCode,
         sharePath: path,
         onRouteClose: () => dispatch({ type: 'CLEAR_CATEGORY', params: { key } })
       },
       newKey: key
-    })
-
-    // $FlowFixMe For some reason action is not allowed to be a StackAction
-    navigation.navigate({
-      routeName: 'CityContent',
-      action: action
-    })
+    }))
```

Comment:
This seems to be wrong... I guess we should dispatch the StackAction in the StackNavigator, not in the top level SwitchNavigator...  (Just an educated guess)

Reply:
I think I did this change some time back as well and it led to problems... not sure of which kind though anymore...

Reply:
It does make sense, but I think it is not possible, as there is no real entry point to a city before you actually enter the city (with specific city and language code). Before you are just somewhere in the switch navigator (in this case in the landing container) which itself is only a basic Navigator.